### ISSUE TITLE
(WCAG 2.1) Lack of Accessible Label for Close Control

### ISSUE THREAD

**Issue by dkatzz** [civiform/civiform#5520]
OBSERVED RESULTS:

- The Close control (dismiss button) on the "Create an Account or Sign In" modal dialog lacks an accessible label. The control is currently implemented as an SVG image, making it unclear to screen reader users.
- ![Screenshot](https://github.com/civiform/civiform/assets/86739416/0bf6881d-4738-49c5-a4cb-82d3f04234e0)

EXPECTED RESULTS:
 
- The Close control on the modal dialog should have a meaningful and accessible label that conveys its purpose to all users, including screen reader users.
- Implement the Close control using an HTML <button> element with an appropriate aria-label attribute. Ensure that the aria-label attribute provides a concise description of the action, such as "Close" or "Dismiss."

STEPS TO REPRODUCE:

- Enable screen reader.
- Navigate to [URL](https://demo.civiform.dev/)
- Navigate to and activate the ‘Apply’ for the Comprehensive sample program.
- Navigate to and activate “What is your address?”.
- Fillup the form fields and complete the application.
- On the confirmation page navigate to and activate “Apply to Another Program" using the tab key.
- Navigate to the “close” button and observe that the button is not keyboard accessible and the screen reader doesn't announce it as a close/dismiss button.

ENVIRONMENT DETAILS:

- Device(s) (brand, OS version, test key or release key): Windows 10 Enterprise (64-bit),MacOS Ventura Version 13.4
- Screen reader name and version: JAWS 2022.2211.7, NVDA 2022.3, VoiceOver
- Browser name and version: Chrome Version 113.0.5672.93
